model_v4.py

```diff
@@ -0,0 +1,382 @@
+import numpy as np
+import os
+from sklearn.metrics.pairwise import pairwise_distances
+from argparse import ArgumentParser
+
+from tensorflow.keras.layers import Input, Dense, \
+                                    Conv2D, MaxPooling2D, \
+                                    AveragePooling2D, Dropout, \
+                                    BatchNormalization, Flatten, \
+                                    Concatenate, GlobalAveragePooling2D, \
+                                    Lambda, Activation
+from tensorflow.keras.regularizers import l2
+from tensorflow.keras.models import Sequential, Model
+from tensorflow.keras.utils import plot_model
+from tensorflow.keras.preprocessing.image import load_img, img_to_array
+from tensorflow.keras.backend import l2_normalize
+
+
+class FaceEncoder():
+    def __init__(self):
+
+        self.order_w_name_idx = (1, 2, 3, 0)
+        self.order_b_name_idx = (1, 0)
+
+        input_size = Input(shape=(96, 96, 3))
+
+        self.conv1 = Conv2D(filters=64, kernel_size=(7, 7), strides=(2, 2),
```

Comment:
You have not to store reference to all layers inside yours class because it will break garbage  collection.